yt_dlp/extractor/tiktok.py

```diff
@@ -511,23 +535,43 @@ def _real_extract(self, url):
         try:
             return self._extract_aweme_app(video_id)
         except ExtractorError as e:
-            self.report_warning(f'{e}; Retrying with webpage')
-
+            self.report_warning(f'{e.orig_msg}; Retrying with webpage')
+
+        def get_random_cookie():
+            """Creates a random cookie (name, value) pair compliant with RFC 6265"""
+            alnum = string.ascii_letters + string.digits
+            # text: printable not in whitespace
+            name_set = '!#$%&\'*+-.^_`|~' + alnum  # in text not in '()<>@,;:"/[]?={}\\'
+            value_set = '!#$%&\'()*+-./:<=>?@[]^_`{|}~' + alnum  # in text not in '",;\\'
+            name_length = random.randint(1, 4096 - 1)  # keep at least one byte for value
+            value_length = random.randint(1, 4096 - name_length)
+            return ''.join(random.choice(name_set) for _ in range(name_length)), \
+                   ''.join(random.choice(value_set) for _ in range(value_length))
+
+        self._set_cookie('www.tiktok.com', *get_random_cookie())
```

Comment:
Is all this code really needed to create a random cookie? Just creating a small random key and value from `ascii_letters` would significantly reduce this code. I assume we don't care about uniform distrinution of the generated cookies.

PS: Why is a random cookie needed anyway?

Reply:
> compliant with RFC 6265

I'm not sure, but it prevent error in #3551. This is still just a workaround, but better than setting UA.
I'm planning to move this in some where else, not sure where.